### PR TITLE
Add debug log if location coming from location provider is not fresh enough

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
Allows developers to understand if they have problems with obsolete locations coming from location provider.

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt [hunk 1]
```diff
@@ -55,6 +56,16 @@ internal class TripSessionLocationEngine constructor(
...
+                // log if location is not fresh enough
+                val currentTime = SystemClock.elapsedRealtimeNanos()
+                val locationAge = (currentTime - it.elapsedRealtimeNanos)
+                if (locationAge > DELAYED_LOCATION_WARNING_THRESHOLD_NS) {
+                    logD(
+                        TAG,
+                        "Got obsolete location: current time = $currentTime, " +
+                            "elapsedRealtimeNanos = ${it.elapsedRealtimeNanos}"
+                    )
+                }
```
Comment (VysotskiVadim) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt:68]: maybe replace comment by a function:
```kotlin
            fun logIfLocationIsNotFreashEnough(location: Location) {
                val currentTime = SystemClock.elapsedRealtimeNanos()
                val locationAge = (currentTime - it.elapsedRealtimeNanos)
                if (locationAge > DELAYED_LOCATION_WARNING_THRESHOLD_NS) {
                    logD(
                        TAG,
                        "Got obsolete location: current time = $currentTime, " +
                            "elapsedRealtimeNanos = ${it.elapsedRealtimeNanos}"
                    )
                }
            }
```
and 
```suggestion
               logIfLocationIsNotFreashEnough(it)
```

**Comment by Guardiola31337** [comment]
Could you also fixup the commit history? 🙏

**Comment by SiarheiFedartsou** [comment]
> Could you also fixup the commit history? 🙏

I can, but do we know why CI takes forever? Firebase seems to be green today.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt [hunk 2]
```diff
@@ -52,9 +53,22 @@ internal class TripSessionLocationEngine constructor(
...
+        val currentTime = SystemClock.elapsedRealtimeNanos()
+        val locationAge = (currentTime - location.elapsedRealtimeNanos)
+        if (locationAge > DELAYED_LOCATION_WARNING_THRESHOLD_NS) {
+            logD(
+                TAG,
+                "Got obsolete location: current time = $currentTime, " +
+                    "elapsedRealtimeNanos = ${location.elapsedRealtimeNanos}"
+            )
+        }
+    }
```
Comment (VysotskiVadim) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt:66]: After some testing I realised that it's hard to read messages in logs. Here's my suggestion:
```suggestion
    private fun logIfLocationIsNotFreshEnough(location: Location) {
        val currentTime = SystemClock.elapsedRealtimeNanos()
        val locationAgeNanoSeconds = currentTime - location.elapsedRealtimeNanos
        val locationAgeMilliseconds = TimeUnit.MILLISECONDS.convert(
            locationAgeNanoSeconds,
            TimeUnit.NANOSECONDS
        )
        if (locationAgeMilliseconds > DELAYED_LOCATION_WARNING_THRESHOLD_MS) {
            logW(TAG, "Got an obsolete location: age = $locationAgeMilliseconds ms")
        }
    }
```

**Comment by VysotskiVadim** [comment]
> Could you also fixup the commit history? 🙏

@Guardiola31337  , is it necessary if he squashes commits?

**Comment by Guardiola31337** [comment]
> is it necessary if he squashes commits?

I personally prefer to have a clean commit history if possible, `bisect` is hard with squashed commits 😬

**Comment by SiarheiFedartsou** [comment]
> > is it necessary if he squashes commits?
> 
> I personally prefer to have a clean commit history if possible, `bisect` is hard with squashed commits 😬

in case of squash it will be only one commit in the main branch, just like as if I fixup it in PR and merge afterwards

the only difference is that history in PR itself will remain the same

**Comment by Guardiola31337** [comment]
> Firebase seems to be green today.

Nah they're still "orange" https://status.firebase.google.com/incidents/3ZP5RThpQqsJxm8a3YKD 😬

**Comment by SiarheiFedartsou** [comment]
> > Firebase seems to be green today.
> 
> Nah they're still "orange" https://status.firebase.google.com/incidents/3ZP5RThpQqsJxm8a3YKD 😬

I.e. we won't be able to merge/release until they fixed this incident?

**Comment by Guardiola31337** [comment]
> I.e. we won't be able to merge/release until they fixed this incident?

If we see that they don't fix the issue, we can definitely go ahead and merge without those checks 👍

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt (hunk 2 above)
Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt:66]: Went ahead and rebase from `main` to fix the conflicts and applied this suggestion. BTW any particular reason to change from `logD` to `logW`?

**Comment by Guardiola31337** [comment]
Going to fixup the commit history and re-push.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt (hunk 2 above)
Comment (SiarheiFedartsou) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt]: I've already applied it manually(left `logD` though).

**Review by Guardiola31337 — APPROVED** [review]
Going ahead and merging here.

Thanks for the contribution @SiarheiFedartsou

**Pull request merged**